shared/js/media/video_player.js

```diff
@@ -161,6 +167,15 @@ function VideoPlayer(container) {
     // The only place we call showPlayer() is from the play() function.
     // If play() has to show the player, call it again when we're ready to play.
     player.oncanplay = function() {
+      if (videowidth != player.videoWidth ||
+          videoheight != player.videoHeight)
+      {
+        // Our guess at the video size was wrong, resize the display
```

Comment:
I expect on some devices or in some recording modes they will be the same size. However this is not guaranteed. The selection algorithm is best effort and somewhat complicated. The best preview size match is an aspect ratio matching (relative to the recoding size) size closest in area to the desired preview size (the window size in our case). If you turn on the SHB, it will also have an impact.

If there is a tangible performance impact, I'm happy to look at upscaling the image over forcing more drastic changes in the application. I'm not quite sure how to do that right now as I'm using lower level APIs than what canvas / 2d context elements expose to the DOM. It just wasn't the first solution that came to mind compared to tweaking the display code :).

I will change it to resize it when the metadata has loaded instead of when it can start playback. Gets it out of the way sooner.